### PR TITLE
Improve parameter translators for NOT_SET cases

### PR DESCRIPTION
**Pull request by domire8**
Let's do the changes required to fix empty parameter declaration and handling in small steps. Here is the first one with the changes in modulo_core

### CODE REVIEW

**Comment by domire8** [comment]
Even though we realized we have to overhaul a few things with the parameters, this PR is still important. It currently fails because it depends on a PR in control libraries:

https://github.com/epfl-lasa/control-libraries/pull/301

File: source/modulo_core/modulo_core/translators/parameter_translators.py [hunk 1]
```diff
@@ -15,7 +16,10 @@ def write_parameter(parameter: sr.Parameter) -> Parameter:
     :raises ParameterTranslationError if the parameter could not be written
     :return: The resulting ROS parameter
     """
-    if parameter.get_parameter_type() == sr.ParameterType.BOOL or \
+    if parameter.is_empty():
+        return Parameter(parameter.get_name(), value=None, type_=Parameter.Type.NOT_SET)
+    elif parameter.get_parameter_type() == sr.ParameterType.BOOL or \
+            parameter.get_parameter_type() == sr.ParameterType.BOOL_ARRAY or \
```
Comment (buschbapti) [inline source/modulo_core/modulo_core/translators/parameter_translators.py:22]: This one is weird as it appears below as well

**Review by buschbapti — DISMISSED** [review]

File: source/modulo_core/src/translators/parameter_translators.cpp [hunk 2]
```diff
@@ -184,15 +188,18 @@ std::shared_ptr<state_representation::ParameterInterface> read_parameter(const r
...
+std::shared_ptr<ParameterInterface> read_parameter_const(
+    const rclcpp::Parameter& ros_parameter, const std::shared_ptr<const ParameterInterface>& parameter
 ) {
   if (ros_parameter.get_name() != parameter->get_name()) {
     throw exceptions::ParameterTranslationException(
         "The ROS parameter " + ros_parameter.get_name()
             + " to be read does not have the same name as the reference parameter " + parameter->get_name());
   }
+  if (ros_parameter.get_type() == rclcpp::PARAMETER_NOT_SET) {
+    return make_shared_parameter_interface(
```
Comment (eeberhard) [inline source/modulo_core/src/translators/parameter_translators.cpp:200]: This function was a nice way to resolve the empty copy / construction for NOT SET 👍

**Review by eeberhard — DISMISSED** [review]
Seems good once it builds, except for the BOOL_ARRAY duplication that @buschbapti caught in parameter_translators.py

**Pull request merged**